src/test/java/mate/academy/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class UserServiceImplTest {
+    private UserService userService;
+    private UserDao userDao;
+    private PasswordEncoder passwordEncoder;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        userDao = Mockito.mock(UserDao.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        userService = new UserServiceImpl(userDao, passwordEncoder);
+        user = new User();
+        user.setEmail("bob@i.ua");
+        user.setPassword("1234");
+    }
+
+    @Test
+    void save_Ok() {
```

Comment:
```suggestion
    void save_ok() {
```
in other tests the same